File: frontend/src/models/Change.ts
import { Entity } from "./Entity";

export default class Change {
    private readonly _id: number;
    private readonly _commitHash: string;
    private readonly _entity: Entity;
    private readonly _churn: number;

    constructor(id: number, commitHash: string, entity: Entity, churn: number) {
        this._id = id;
        this._commitHash = commitHash;
        this._entity = entity;
        this._churn = churn;
    }

    public get id() {
        return this._id;
    }

    public get commitHash() {
        return this._commitHash;
    }

    public get entity() {
        return this._entity;
    }

    public get churn() {
        return this._churn;
    }
}

File: seacap-extractor/src/main/java/net/jlefever/seacap/dump/models/Change.java
package net.jlefever.seacap.dump.models;

public class Change
{
    private final int id;
    private final int entityId;
    private final String commitHash;
    private final int churn;

    public Change(int id, int entityId, String commitHash, int churn)
    {
        this.id = id;
        this.entityId = entityId;
        this.commitHash = commitHash;
        this.churn = churn;
    }

    public int getId()
    {
        return id;
    }

    public int getEntityId()
    {
        return entityId;
    }

    public String getCommitHash()
    {
        return commitHash;
    }

    public int getChurn()
    {
        return churn;
    }
}
